Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -132,7 +137,13 @@ public final class CoreDataManager: StorageManagerType {
     /// Saves the derived storage. Note: the closure may be called on a different thread
     ///
     public func saveDerivedType(derivedStorage: StorageType, _ closure: @escaping () -> Void) {
-        derivedStorage.perform {
+        derivedStorage.perform { [weak self] in
+            guard let self = self else {
+                return
+            }
+            guard derivedStorage.name == self.contextName else {
+                return
+            }
```

Comment:
I was hoping that we could check a property (like [`persistentStoreCoordinator`](https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/1506618-persistentstorecoordinator)). I'm curious if you've tried that? There doesn't seem to be a property like `isValid`. 🤷

Reply:
yea, I tried comparing the child context's `persistentStoreCoordinator` and its persistent stores with the main context's, but they seem to be always the same because it's a singleton (`CoreDataManager`'s `persistentContainer.persistentStoreCoordinator`) and I didn't find any Core Data API to check on if a context / persistent store has been reset 😞 

It's quite frustrating how little official guidance there is for Core Data best practices for different scenarios (app reset, parent/child context etc.).